### DISCUSSION TITLE
Why can't `remove` be destructured from a job?

### DISCUSSION THREAD

**Discussion by asvinp** [taskforcesh/bullmq#2456] [Q&A]
I have this code where onComplete or onError I remove the job. If I destructure `remove` from the job and call it, it would fail with a typeerror (see below), where as if I don't destructure it works as expected. 

Appreciate if someone could help explain why this happens. 

```
import { Job } from 'bullmq'

completed: async (job: Job) => {
      const { id, name, data, returnvalue } = job

        await job.remove() // This works
}
```

```
import { Job } from 'bullmq'

completed: async (job: Job) => {
      const { id, name, data, returnvalue, remove } = job

        await remove() // This fails
}
```

Error: `TypeError: Cannot read properties of undefined (reading 'queue')`

**Comment by manast** [comment]
This is how javascript works actually. As you are calling remove as a standalone function, but this function was part of a class before, you lost the correct "this" pointer.  TLDR; do not destructure class methods it will never work.
